proto/storage/config.proto

```diff
@@ -79,9 +79,22 @@ message PrivateConfig {
   AdministrationEventsConfig administration_events_config = 8;
 }
 
+message PlatformComponentConfig {
+  message Rule {
+    message NamespaceRule {
+      string regex = 1;
+    }
+    string name = 1;
+    NamespaceRule namespace_rule = 2;
+  }
+  repeated Rule rules = 1;
+  bool needs_reevaluation = 2;
```

Comment:
Feels odd to have such a thing like this in the storage proto.  Is that because the storage object is sadly also the API object?

Reply:
Seems like a reasonable thing to discuss going forward.  F2F topic?  Especially if there are significant changes like this one.

Reply:
Yeah I did not add a v2 API for this one because the UI would continue to use the PUT v1/config to update the platform component config. That same API is also used to update retention configs and I did not want to migrate all the configs to v2 for this one.